crypto/cryptlib.c

```diff
@@ -396,15 +396,34 @@ void OPENSSL_showfatal(const char *fmta, ...)
 # endif
 }
 #else
+
+# ifdef __APPLE__
+#  include <Availability.h>
+#   if (defined(__MAC_10_11)  && __MAC_OS_X_VERSION_MIN_REQUIRED  >= __MAC_10_11) || \
+       (defined(__IPHONE_9_0) && __IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_9_0)
```

Comment:
We've had problems in the past with some compilers falling over with expressions of this form:

````
#if defined(MYMACRO) && MYMACRO > 0
````

The problem being that if the macro is not defined they do not short circuit and still attempt to evaluate the second part of the expression.

Reply:
Removed these ifdefs in favor of a runtime check for a weak symbol.